modules/eks/argocd/README.md

```diff
@@ -14,14 +14,132 @@ kubectl apply -k "https://github.com/argoproj/argo-cd/manifests/crds?ref=v2.4.9"
+that includes all the details for helm to create ArgoCD CRDs. These CRDs
+let ArgoCD know how to fulfill changes to its repo.
+
+```yaml
+components:
+  terraform:
+    argocd-repo-defaults:
+      metadata:
+        type: abstract
+      vars:
+        enabled: true
+        github_user: acme_admin
+        github_user_email: infra@acme.com
+        github_organization: ACME
+        github_codeowner_teams:
+        - "@ACME/acme-admins"
+        - "@ACME/CloudPosse"
+        - "@ACME/developers"
+        gitignore_entries:
+          - "**/.DS_Store"
+          - ".DS_Store"
+          - "**/.vscode"
+          - "./vscode"
+          - ".idea/"
+          - ".vscode/"
+        permissions:
+          - team_slug: acme-admins
+            permission: admin
+          - team_slug: CloudPosse
+            permission: admin
+          - team_slug: developers
+            permission: push
+```
+
+### Injecting infrastructure details into applications
+Second, your application repos could use values to best configure their
+helm releases. We have an `eks/platform` component for exposing various
+infra outputs. It takes remote state lookups and stores them into SSM.
+We demonstrate how to pull the platform SSM parameters later. Here's an
+example `eks/platform` config:
+
+```yaml
+components:
+  terraform:
+    eks/platform:
+      metadata:
+        type: abstract
+        component: eks/platform
+      backend:
+        s3:
+          workspace_key_prefix: platform
+      deps:
+        - catalog/eks/cluster
+        - catalog/eks/alb-controller-ingress-group
+        - catalog/acm
+      vars:
+        enabled: true
+        name: "platform"
+        eks_component_name: eks/cluster
+        ssm_platform_path: /platform/%s/%s
+        references:
+          default_alb_ingress_group:
+            component: eks/alb-controller-ingress-group
+            output: .group_name
+          default_ingress_domain:
+            component: dns-delegated
+            environment: gbl
+            output: "[.zones[].name][-1]"
+
+    eks/platform/acm:
+      metadata:
+        component: eks/platform
+        inherits:
+          - eks/platform
+      vars:
+        eks_component_name: eks/cluster
+        references:
+          default_ingress_domain:
+            component: acm
+            environment: use2
+            output: .domain_name
+
+    eks/platform/dev:
+      metadata:
+        component: eks/platform
+        inherits:
+          - eks/platform
+      vars:
+        platform_environment: dev
+
+    acm/qa2:
+      settings:
+        spacelift:
+          workspace_enabled: true
+      metadata:
+        component: acm
+      vars:
+        enabled: true
+        name: acm-qa2
+        tags:
+          Team: sre
+          Service: acm
+        process_domain_validation_options: true
+        validation_method: DNS
+        dns_private_zone_enabled: false
+        certificate_authority_enabled: false
```

Comment:
```suggestion
```
no need for acm example in "injecting TF into app code" section

Reply:
The platform environments take ACM components. We provide them here because otherwise the application will be missing certificates, and we've seen that ingress controllers won't set up without those working. Not only that, it's difficult to ascertain that it's the problem. So the absence of this detail hides effort and perpetuates a difficult hurdle.